src/utils/daos_hdlr.c

```diff
@@ -2062,26 +2063,21 @@ mkdir_dfs(struct cmd_args_s *ap, struct file_dfs *file_dfs, const char *path,
 		D_GOTO(out, rc = EINVAL);
 	}
 
-	rc = dfs_lookup(file_dfs->dfs, dname,
-			O_RDWR, &parent, NULL, NULL);
-	if (parent == NULL) {
-		fprintf(ap->errstream, "dfs_lookup %s failed\n", dname);
-		D_GOTO(out, rc = EINVAL);
+	rc = dfs_lookup(file_dfs->dfs, dname, O_RDWR, &parent, NULL, NULL);
+	if (rc != 0) {
+		DH_PERROR_SYS(ap, rc, "dfs_lookup '%s' failed", dname);
+		D_GOTO(out, rc);
 	}
 	rc = dfs_mkdir(file_dfs->dfs, parent, name, *mode, 0);
 	if (rc != 0) {
-		/* continue if directory exists, fail otherwise */
-		fprintf(ap->errstream, "dfs_mkdir %s failed, %s\n",
-			name, strerror(rc));
+		/* TODO: continue if directory exists, fail otherwise */
```

Comment:
Actually, the caller handles EEXIST, so I think we can just remove this comment entirely. Nice catch!
https://github.com/daos-stack/daos/blob/f4f9742e8bece23a65888ae91c8cd05e39bc7acd/src/utils/daos_hdlr.c#L2558-L2560

Reply:
I have to say, this comment didn't make much sense to me at all.

There are quite a few places in this code where the wrong error is reported, most of which I haven't tried to address, either because there isn't a returncode from static functions or errno is being using weirdly.

Reply:
Luckily, many of these will be handled when the code is updated to use the dfs_sys API